src/Microsoft.OpenApi/Extensions/OpenApiReferencableExtensions.cs

```diff
@@ -54,12 +54,12 @@ public static IOpenApiReferenceable ResolveReference(this IOpenApiReferenceable
 
         private static IOpenApiReferenceable ResolveReferenceOnHeaderElement(
             OpenApiHeader headerElement,
-            string propertyName,
-            string mapKey,
+            string? propertyName,
+            string? mapKey,
             JsonPointer pointer)
         {
             if (OpenApiConstants.Examples.Equals(propertyName, StringComparison.Ordinal) &&
-                !string.IsNullOrEmpty(mapKey) &&
+                mapKey is not null &&
```

Comment:
you are changing the behaviour but not checking for empty string anymore, this should be reverted (and other places in this document)

Reply:
string.IsNullOrEmpty doesn't handle nullable strings properly, mapKey is still treated as string? once it exits the condition leading to compiler warnings in the subsequent code.
I can only think of using the null-forgiving operator in conjuction with `string.IsNullOrEmpty` otherwise we might have to explicitly check whether a string is null or empty as two separate checks.
Do you have a better approach?

Reply:
> string.IsNullOrEmpty doesn't handle nullable strings properly, mapKey is still treated as string? once it exits the condition leading to compiler warnings in the subsequent code.

That's only because we're targeting netstandard2 and the stdlib does NOT have the nullability annotations. If we targeted net8 only (not suggesting we should change that), you wouldn't have compiler warnings anymore.

> Do you have a better approach?

not really, I think we could have both (is not null + string null or empty) to make the compiler happy and maintain the current behaviour.

What do you think?

Reply:
Yeah I think this suffices for now.